File: VolleyballApp/VolleyballApp.API/Models/Invite.cs
namespace VolleyballApp.API.Models
{
    public class Invite
    {
        public int Id { get; set; }
        public User InviteFrom { get; set; }
        public User InviteTo { get; set; }
        public bool FriendInvite { get; set; }
        public bool TeamInvite { get; set; }
        public bool MatchInvite { get; set; }
    }
}

File: VolleyballApp/VolleyballApp-SPA/src/app/_models/invite.ts
import { Match } from './match';
import { Team } from './team';
import { User } from './user';

export interface Invite {
    id: number;
    inviteFrom?: User;
    inviteTo?: User;
    teamInviting?: Team;
    teamInvited?: Team;
    matchInvitedTo: Match;
    friendInvite: boolean;
    teamInvite: boolean;
    matchInvite: boolean;
    refereeInvite: boolean;
}
